File: sdk/nodejs/src/management.ts
import { AxiosInstance } from 'axios';

export interface UserResponse {
  id: string;
  email: string;
  full_name?: string;
  company?: string;
  plan: string;
  credit_balance: number;
  is_active: boolean;
  created_at: string;
}

export interface APIKeyResponse {
  id: string;
  name: string;
  key?: string;
  key_prefix: string;
  permissions: string[];
  rate_limit: number;
  is_active: boolean;
  created_at: string;
  last_used?: string;
}

export interface UsageStats {
  total_requests: number;
  total_cost: number;
  total_tokens: number;
  by_endpoint: Record<string, any>;
  period: string;
}

export class ManagementAPI {
  constructor(private client: AxiosInstance) {}

  /**
   * Get current user information
   */
  async getCurrentUser(): Promise<UserResponse> {
    const response = await this.client.get('/api/v1/manage/users/me');
    return response.data;
  }

  /**
   * Create new API key
   */
  async createAPIKey(name: string, permissions?: string[]): Promise<APIKeyResponse> {
    const response = await this.client.post('/api/v1/manage/api-keys', {
      name,
      permissions: permissions || ['*']
    });
    return response.data;
  }

  /**
   * List all API keys
   */
  async listAPIKeys(): Promise<APIKeyResponse[]> {
    const response = await this.client.get('/api/v1/manage/api-keys');
    return response.data;
  }

  /**
   * Revoke API key
   */
  async revokeAPIKey(keyId: string): Promise<void> {
    await this.client.delete(`/api/v1/manage/api-keys/${keyId}`);
  }

  /**
   * Get usage statistics
   */
  async getUsage(period: string = '30d'): Promise<UsageStats> {
    const response = await this.client.get('/api/v1/manage/usage', {
      params: { period }
    });
    return response.data;
  }

  /**
   * Add credits to account
   */
  async addCredits(amount: number): Promise<any> {
    const response = await this.client.post('/api/v1/manage/credits', {
      amount
    });
    return response.data;
  }

  /**
   * Update subscription plan
   */
  async updateSubscription(plan: string): Promise<any> {
    const response = await this.client.post(`/api/v1/manage/subscription/${plan}`);
    return response.data;
  }
}

File: sdk/python/money_api/management.py
"""Management API module."""
from typing import Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from money_api.client import MoneyAPI


class ManagementAPI:
    """Account management operations."""

    def __init__(self, client: "MoneyAPI"):
        self.client = client

    def get_usage(self, period: str = "30d") -> Dict:
        """Get usage statistics.

        Args:
            period: Time period (24h, 7d, 30d, 90d)

        Returns:
            Usage statistics
        """
        return self.client.request(
            "GET",
            "/api/v1/manage/usage",
            params={"period": period}
        )

    def get_current_user(self) -> Dict:
        """Get current user information.

        Returns:
            User account details
        """
        return self.client.request("GET", "/api/v1/manage/users/me")

    def create_api_key(self, name: str) -> Dict:
        """Create new API key.

        Args:
            name: Name for the API key

        Returns:
            New API key details
        """
        return self.client.request(
            "POST",
            "/api/v1/manage/api-keys",
            json={"name": name}
        )

    def list_api_keys(self) -> Dict:
        """List all API keys.

        Returns:
            List of API keys
        """
        return self.client.request("GET", "/api/v1/manage/api-keys")
